src/main/java/gift/product/entity/Product.java

```diff
@@ -87,14 +87,12 @@ public void validateOptionForUpdate(String name, Long optionId) {
         checkDuplicateOptionName(name, optionId);
     }
 
-    public void decreaseOptionQuantity(Long optionId, int quantity) {
-        Option option = getOptionByOptionId(optionId);
+    public void decreaseOptionQuantity(Option option, int quantity) {
+        if(!this.getId().equals(option.getProduct().getId())) {
```

Comment:
```suggestion
        if(!this.id.equals(option.getProduct().getId())) {
```
클래스 안에 있어서 직접 접근할 수 있을 것 같아요!

Reply:
피드백 감사합니다!!